src/dataflow/utilities/node.ts

```diff
@@ -217,6 +234,42 @@ export const NodeGeneratorTypes = [
   */
 ];
 
+export const NodeGeneratorPeriodUnits = [
+  {
+    unit: "sec",
+    lengthInSeconds: 1
+  },
+  {
+    unit: "min",
+    lengthInSeconds: 60
+  },
+  {
+    unit: "hour",
+    lengthInSeconds: 3600
+  },
+];
+
+export const NodeTimerInfo =
+{
+  name: "Timer",
+  method: (t: number, p: number, a: number) => t % p < p / 2 ? 1 * a : 0,
+};
+
+export const NodeTimerPeriodUnits = [
+  {
+    unit: "sec",
+    lengthInSeconds: 1
+  },
+  {
+    unit: "min",
+    lengthInSeconds: 60
+  },
+  {
+    unit: "hour",
+    lengthInSeconds: 3600
+  },
+];
+
```

Comment:
At present the generator and timer nodes use the same units.  However, the generator node has changed frequently in the past few months so I'm anticipating that these might eventually diverge.  That said, I wanted to separate them now,